pandas/tests/extension/test_integer.py

```diff
@@ -238,9 +238,12 @@ def check_reduce(self, s, op_name, skipna):
         # overwrite to ensure pd.NA is tested instead of np.nan
         # https://github.com/pandas-dev/pandas/issues/30958
         result = getattr(s, op_name)(skipna=skipna)
-        expected = getattr(s.astype("float64"), op_name)(skipna=skipna)
-        if np.isnan(expected):
+        if skipna:
+            expected = getattr(s.dropna().astype("int64"), op_name)(skipna=skipna)
```

Comment:
+1, that seems better ;)

Reply:
But actually, if there are no NAs, the dropna of this line is also fine. So I think you can do:

if not skipna and s.isna().any():
    expected = pd.NA
else:
    expected = ... (with s.dropn().astype(int64) ..

Reply:
Nice 👍